### ISSUE TITLE
Support Influxdb

### ISSUE THREAD

**Issue by JulienChampseix** [arachnys/cabot#151]
Hi Team,

that's could be usefull to have the influxdb support for metrics part.

Is it something already think or planned ?

Thanks for your feedback.
Julien

**Comment by dever860** [comment]
+1

**Comment by hanej** [comment]
+2

**Comment by JulienChampseix** [comment]
@bonniejools any news ?

**Comment by rishid** [comment]
It would probably make sense to wait for InfluxDb v0.9 as the query language is going to change significantly. It is due in the next couple weeks.

**Comment by bonniejools** [comment]
@JulienChampseix [Alert Plugins](https://github.com/arachnys/cabot/pull/167) are nearing completion and once they do we will be working on checks. I'll give an update once we start working on them.

I assume that influxdb would be a 'check' rather than an alert so it will have to wait. Though given that several others have displayed interest in this feature, it is definitely on the todo list.

**Comment by JulienChampseix** [comment]
Perfect, hope to test them soon for influxdb part. Have you an estimation when that could be ready ?

**Comment by bonniejools** [comment]
@JulienChampseix @rishid @hanej @dever860 Good news folks. Alert plugins were merged to master in #167 so I'll be focussing on check plugins. I'll keep you updated in a new issue as I go on.

**Comment by rishid** [comment]
Cool, you probably want to wait for influxdb v0.9 coming out next week which has breaking API changes. It shouldn't be too different.

**Comment by JulienChampseix** [comment]
@bonniejools great news ! I gonna test the master branch and will provide you feedback soon.

**Comment by rishid** [comment]
One that would be awesome if the new check design could support querying from different data sources. Use case: I have influxdb running in two data centers but would love to only maintain a single cabot instance that can hit both.

**Comment by bonniejools** [comment]
@rishid Sounds like a great idea. I'll keep it in mind.

**Comment by rishid** [comment]
fyi new the v0.9 api docs are available http://influxdb.com/docs/v0.9/concepts/reading_and_writing_data.html

**Comment by JulienChampseix** [comment]
@bonniejools did you progress on check plugins ? Maybe an http check supporting json data as parameter could work

**Comment by johnaschroeder** [comment]
+1

**Comment by benroeder** [comment]
Would be interested in this as well
I know it's nasty, but it would be interesting to be able to use Existing plugins, 
https://github.com/harisekhon/nagios-plugins
Or 
https://exchange.icinga.org

https://www.monitoring-plugins.org

https://github.com/munin-monitoring/contrib/tree/master/plugins

Etc
Ben

**Comment by xinity** [comment]
I don't think this is a good idea. It would stick cabot with the crappy plugin architecture of nagios forks.

**Comment by JulienChampseix** [comment]
agree with @xinity . Hope @bonniejools will provide us the check plugin refactoring soon :)

**Comment by xinity** [comment]
A check plugin architecture would be awesome for sure, Can't wait to Play With it

**Comment by Melevir** [comment]
Gentlemens, what is the status of influxdb support?
I'll be glad to help.

**Comment by JulienChampseix** [comment]
+1 @bonniejools

**Comment by JulienChampseix** [comment]
Up

**Comment by justin8** [comment]
+1 when 0.9 finally hits release.

**Comment by JulienChampseix** [comment]
@bonniejools any news or update about this feature ?

**Comment by cemo** [comment]
+1

**Comment by mahendra** [comment]
+1

**Comment by yooshang** [comment]
+1

**Comment by mahendra** [comment]
I have modified Cabot to support InfluxDB (0.8). Am working on supporting 0.9 now. Will release it sometime next week.

The InfluxDB (0.8) version is available at - https://github.com/Affirm/cabot

It is a crude hack. I am still working on making Cabot support data plugins. Will help in switching data sources at any time. Once I have it, I will send a pull request to the main Cabot branch.

**Comment by egoldblum** [comment]
+1

**Comment by Dentaku1992** [comment]
+1

**Comment by xiemeilong** [comment]
+1

**Comment by mabushey** [comment]
+1

**Comment by mahendra** [comment]
Just checking if anyone tried - https://github.com/Affirm/cabot for influxdb. been working for us quite well.

**Comment by mabushey** [comment]
@mahendra - What version of influxdb? I don't see a single thing regarding influxDB. :( Is this documented anywhere?

**Comment by mahendra** [comment]
@mabushey we are using influx 0.8. I haven't updated the docs yet. Install, set the following in env/config - https://github.com/Affirm/cabot/blob/master/cabot/cabot_config.py#L16-L19 and have a go

![image](https://cloud.githubusercontent.com/assets/115988/10088050/d086868e-62cc-11e5-81ff-fa64c5a1d020.png)

**Comment by JulienChampseix** [comment]
@mahendra does it support influxdb 0.9 ?

**Comment by mahendra** [comment]
nope. not yet. we are planning on upgrading to 0.9 .. maybe in a month or two

**Comment by blysik** [comment]
We're investigating moving to influxdb sometime in the next year, so this is an interesting feature to me too!

**Comment by ddes** [comment]
Hi,

+1, I'm very interrested too, especially 0.9 support.

Any news (official or not) on this ?

**Comment by guilleJB** [comment]
Hi,

I'm interested too. We are using influx 0.9

**Comment by NuckChorris** [comment]
+1 I'm looking at deploying InfluxDB and would love to have Cabot hooked to it

**Comment by mahendra** [comment]
@NuckChorris we are using InfluxDB 0.8. Not too happy with the stability of Influx 0.9. Cabot support for influx 0.9 is on our pipeline. Should be out in 2 months. Will publish it here - https://github.com/Affirm/cabot
